src/dotenv/main.py

```diff
@@ -226,14 +273,17 @@ def unset_key(dotenv_path, key_to_unset, quote_mode="always"):
                 dest.write(mapping.original)
 
     if not removed:
-        warnings.warn("key %s not removed from %s - key doesn't exist." % (key_to_unset, dotenv_path))
+        if not TYPE_CHECKING or sys.version_info >= (3, 0):
```

Comment:
I think this obscures the code too much.  I would probably just ignore the type warning for the line below.  This applies to other uses of `warnings.warn`.